src/scripts/main.js

```diff
@@ -1,3 +1,188 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const table = document.querySelector('table');
+  const headers = table.querySelectorAll('th');
+  const tbody = table.querySelector('tbody');
+
+  let currentSortColumnIndex = null;
+  let isAscending = true;
+
+  headers.forEach((header, index) => {
+    header.addEventListener('click', () => {
+      if (currentSortColumnIndex === index) {
+        isAscending = !isAscending;
+      } else {
+        currentSortColumnIndex = index;
+        isAscending = true;
+      }
+
+      const rows = Array.from(tbody.querySelectorAll('tr'));
+
+      rows.sort((rowA, rowB) => {
+        let cellA = rowA.children[index].textContent.trim();
+        let cellB = rowB.children[index].textContent.trim();
+
+        if (index === 4) {
+          cellA = parseFloat(cellA.replace(/[$,]/g, '')) || 0;
+          cellB = parseFloat(cellB.replace(/[$,]/g, '')) || 0;
+        }
+
+        const isNumeric = !isNaN(cellA) && !isNaN(cellB);
+
+        if (isNumeric) {
+          return isAscending ? cellA - cellB : cellB - cellA;
+        } else {
+          return isAscending
+            ? cellA.localeCompare(cellB)
+            : cellB.localeCompare(cellA);
+        }
+      });
+
+      tbody.innerHTML = '';
+      rows.forEach((row) => tbody.appendChild(row));
+    });
+  });
+
+  tbody.addEventListener('click', (e) => {
+    const clickedRow = e.target.closest('tr');
+
+    if (clickedRow) {
+      const activeRow = tbody.querySelector('.active');
+
+      if (activeRow) {
+        activeRow.classList.remove('active');
+      }
+      clickedRow.classList.add('active');
+    }
+  });
+
+  const form = document.createElement('form');
+
+  document.body.appendChild(form);
+  form.className = 'new-employee-form';
+
+  const fields = [
+    {
+      label: 'Name',
+      name: 'name',
+      type: 'text',
+      dataQa: 'name',
+    },
+    {
+      label: 'Position',
+      name: 'position',
+      type: 'text',
+      dataQa: 'position',
+    },
+    {
+      label: 'Age',
+      name: 'age',
+      type: 'number',
+      dataQa: 'age',
+    },
+    {
+      label: 'Salary',
+      name: 'salary',
+      type: 'number',
+      dataQa: 'salary',
+    },
+  ];
+
+  fields.forEach(({ label, name: n, type, dataQa }) => {
+    const fieldLabel = document.createElement('label');
+
+    fieldLabel.innerHTML = `${label}: <input name="${n}" type="${type}" data-qa="${dataQa}" required>`;
+    form.appendChild(fieldLabel);
+  });
+
+  const officeLabel = document.createElement('label');
+
+  officeLabel.innerHTML = `Office: <select name="office" data-qa="office" required>
```

Comment:
The HTML string for the select element is not properly closed. The backtick (`) should be moved to the end of the select element, after the closing `</select>` tag.